tests/MockApi.php

```diff
@@ -0,0 +1,154 @@
+<?php
+namespace Auth0\Tests;
+
+use Auth0\SDK\API\Management;
+use Auth0\SDK\API\Management\Blacklists as BL;
+use Auth0\SDK\API\Management\Clients as Cl;
+use Auth0\SDK\API\Management\ClientGrants as CG;
+use Auth0\SDK\API\Management\Connections as Co;
+use Auth0\SDK\API\Management\DeviceCredentials as DC;
+use Auth0\SDK\API\Management\Emails as Em;
+use Auth0\SDK\API\Management\EmailTemplates as ET;
+use Auth0\SDK\API\Management\Jobs as J;
+use Auth0\SDK\API\Management\Logs as L;
+use Auth0\SDK\API\Management\ResourceServers as RS;
+use Auth0\SDK\API\Management\Rules as R;
+use Auth0\SDK\API\Management\Stats as S;
+use Auth0\SDK\API\Management\Tenants as Te;
+use Auth0\SDK\API\Management\Tickets as Ti;
+use Auth0\SDK\API\Management\UserBlocks as UB;
+use Auth0\SDK\API\Management\Users as U;
+use Auth0\SDK\API\Management\UsersByEmail as UE;
+use GuzzleHttp\Handler\MockHandler;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\Middleware;
+use GuzzleHttp\Psr7\Request;
+
+/**
+ * Class MockApi
+ *
+ * @package Auth0\Tests
+ */
+class MockApi
+{
+
+    /**
+     * Base API URL for the mock API.
+     */
+    const API_BASE_URL = 'https://api.test.local/api/v2/';
+
+    /**
+     * Guzzle request history container for mock API.
+     *
+     * @var array
+     */
+    protected $requestHistory = [];
+
+    /**
+     * Management API object.
+     *
+     * @var Management
+     */
+    protected $client;
+
+    /**
+     * Management API endpoint name.
+     *
+     * @var string
+     */
+    protected $endpoint;
+
+    /**
+     * MockApi constructor.
+     *
+     * @param string $endpoint  Endpoint property name to call.
+     * @param array  $responses Responses to be loaded, an array of Response objects.
+     */
+    public function __construct($endpoint, array $responses = [])
```

Comment:
consider instead of receiving an array of responses here, having a method like `enqueue(response)` that would add a single response or an array of responses to the api queue. This way you could instantiate just once the `MockApi` at the setup step of the test file, since the endpoint is common to the test file, and then just enqueue/write and take/read responses. I think it would simplify a lot the file and would also be more reusable.
You could as well forget about indexes and always keep the "last response". e.g. public api:

```js
MockApi mockApi = new MockApi(endpoint);
mockApi.enqueue(response1);
mockApi.enqueue(response2);

response1 = mockApi.take();
mockApi.lastUrl() == response1.url;

response2 = mockApi.take();
mockApi.lastUrl() == response2.url;

mockApi.take(); //throws error, queue empty
```